lte/gateway/python/magma/subscriberdb/store/sqlite.py

```diff
@@ -48,21 +55,111 @@ def _create_store(self):
         with self.conn:
             self.conn.execute("CREATE TABLE IF NOT EXISTS subscriberdb"
                               "(subscriber_id text PRIMARY KEY, data text)")
+        with self.conn:
+            self.conn.execute(
+                "CREATE TABLE IF NOT EXISTS apndb"
+                "(apn_name text PRIMARY KEY, data text, subscriberids text)"
+            )
+
+    def _add_sub_to_apndb(self, subscriber_data):
+        """
+        Method that adds the subscriber ids to apndb.
+        """
+        sid = SIDUtils.to_str(subscriber_data.sid)
+
+        for idx in range(len(subscriber_data.non_3gpp.apn_config)):
+            # Add the subscriber ID in apnDB
+            sids = SubscriberIDSet()
+            res = self.conn.execute(
+                "SELECT subscriberids FROM apndb WHERE " "apn_name = ?",
+                (subscriber_data.non_3gpp.apn_config[idx].service_selection,),
+            )
+            row = res.fetchone()
+            if not row:
+                raise ApnNotFoundError()
+            stored_sid_data = SubscriberIDSet()
+            if row[0]:
+                stored_sid_data.ParseFromString(row[0])
+
+                # Repopulate sids
+                # Add the sids already present in stored_sid_data
+                for sid_idx in range(len(stored_sid_data.sids)):
+                    sid_data = sids.sids.add()
+                    sid_data.id = stored_sid_data.sids[sid_idx].id
+            # Now add the new sids received from test script/cli
+            sid_data = sids.sids.add()
+            sid_data.id = sid
+            data = sids
+            data_str = data.SerializeToString()
+            res = self.conn.execute(
+                "UPDATE apndb SET subscriberids = ? " "WHERE apn_name = ?",
+                (
+                    data_str,
+                    subscriber_data.non_3gpp.apn_config[idx].service_selection,
+                ),
+            )
+
+    def _fill_apn_from_apndb(self, non_3gpp, subscriber_data):
+        """
+        Method that retrieves apn data from apndb and updates the subscriberdb.
+        """
+        # Fetch the APN data from apndb and add to subscriberdb
+        for idx in range(len(subscriber_data.non_3gpp.apn_config)):
```

Comment:
From Tariq: 

You dont need the index. Maybe use
`for apn in subscriber_data.non_3gpp.apn_config:`
The same for many other places